jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java

```diff
@@ -226,7 +226,7 @@ public Object deserialize(Writable blob) throws SerDeException {
         case TIMESTAMP:
           if (rowVal instanceof java.sql.Timestamp) {
             java.sql.Timestamp timestampRowVal = (java.sql.Timestamp) rowVal;
-            rowVal = Timestamp.ofEpochMilli(timestampRowVal.getTime(), timestampRowVal.getNanos());
+            rowVal = new Timestamp(timestampRowVal);
```

Comment:
It seems `Date` type should have same issue (visible depending on time zone). Could we change that accordingly too?

Reply:
Done